txmongo/collection.py

```diff
@@ -181,39 +181,44 @@ def _normalize_fields_projection(fields):
     def _gen_index_name(keys):
         return u'_'.join([u"%s_%s" % item for item in keys])
 
-    @defer.inlineCallbacks
     def _list_collections_3_0(self):
-        response = yield self._database.command(SON([("listCollections", 1),
-                                                     ("filter", {"name": self.name})]))
-        assert response["cursor"]["id"] == 0
-        first_batch = response["cursor"]["firstBatch"]
-        if first_batch:
-            defer.returnValue(first_batch[0])
-        else:
-            defer.returnValue(None)
+        def on_ok(response):
+            assert response["cursor"]["id"] == 0
+            first_batch = response["cursor"]["firstBatch"]
+            if first_batch:
+                return first_batch[0]
+            else:
+                return None
+
+        return self._database.command(SON([("listCollections", 1),
+                                     ("filter", {"name": self.name})]))\
+                   .addCallback(on_ok)
```

Comment:
For readability, can you do a carriage return before SON to get it to the next line? Will this give us enough room to attach .addCallback(on_ok) without having to append an \  ?